moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp

```diff
@@ -364,48 +364,61 @@ ompl::base::ConstraintPtr createOMPLConstraints(const moveit::core::RobotModelCo
                                                 const std::string& group,
                                                 const moveit_msgs::msg::Constraints& constraints)
 {
-  // TODO(bostoncleek): does this reach the end w/o a return ?
-
-  const std::size_t num_dofs = robot_model->getJointModelGroup(group)->getVariableCount();
-  const std::size_t num_pos_con = constraints.position_constraints.size();
-  const std::size_t num_ori_con = constraints.orientation_constraints.size();
-
   // This factory method contains template code to support position and/or orientation constraints.
   // If the specified constraints are invalid, a nullptr is returned.
   std::vector<ompl::base::ConstraintPtr> ompl_constraints;
-  if (num_pos_con > 1)
-  {
-    RCLCPP_WARN(LOGGER, "Only a single position constraint is supported. Using the first one.");
-  }
-  if (num_ori_con > 1)
-  {
-    RCLCPP_WARN(LOGGER, "Only a single orientation constraint is supported. Using the first one.");
-  }
-  if (num_pos_con > 0)
+  const std::size_t num_dofs = robot_model->getJointModelGroup(group)->getVariableCount();
+
+  // Parse Position Constraints
+  if (!constraints.position_constraints.empty())
   {
-    BaseConstraintPtr pos_con;
-    if (constraints.name == "use_equality_constraints")
+    if (constraints.position_constraints.size() > 1)
+    {
+      RCLCPP_WARN(LOGGER, "Only a single position constraint is supported. Using the first one.");
+    }
+
+    const auto& primitives = constraints.position_constraints.at(0).constraint_region.primitives;
+    if (primitives.empty() || primitives.at(0).type != shape_msgs::msg::SolidPrimitive::BOX)
```

Comment:
Based on how the `BoxConstraint` and `EqualityPositionConstraint` parsing works, I think we maybe also want a warning if the number of primitives is greater than 1?